go/teams/transactions.go

```diff
@@ -319,6 +315,55 @@ func (tx *AddMemberTx) CompleteSocialInvitesFor(ctx context.Context, uv keybase1
 	return nil
 }
 
+func (tx *AddMemberTx) completeAllKeybaseInvitesForUID(uv keybase1.UserVersion) error {
+	// Find the right payload first
+	payload := tx.findChangeReqForUV(uv)
+	if payload == nil {
+		return fmt.Errorf("could not find uv %v in transaction", uv)
+	}
+
+	team := tx.team
+	for _, invite := range team.chain().inner.ActiveInvites {
+		if inviteUv, err := invite.KeybaseUserVersion(); err == nil {
+			if inviteUv.Uid.Equal(uv.Uid) {
+				if payload.CompletedInvites == nil {
+					payload.CompletedInvites = make(map[keybase1.TeamInviteID]keybase1.UserVersionPercentForm)
+				}
+				payload.CompletedInvites[invite.Id] = uv.PercentForm()
+			}
+		}
+	}
+
+	return nil
+}
+
+func (tx *AddMemberTx) ReAddMemberToImplicitTeam(ctx context.Context, uv keybase1.UserVersion, hasPUK bool, role keybase1.TeamRole) error {
+	if !tx.team.IsImplicit() {
+		return errors.New("Trying to use ReAddMemberToImplicitTeam on a team that is not implicit.")
+	}
+
+	if hasPUK {
+		tx.addMember(uv, role)
+		tx.sweepCryptoMembers(uv.Uid)
+		if err := tx.completeAllKeybaseInvitesForUID(uv); err != nil {
+			return err
+		}
+	} else {
+		tx.createInvite(uv, role)
+		tx.sweepKeybaseInvites(uv.Uid)
+		// We cannot sweep crypto members here because we need to
+		// ensure that we are only posting one link, and if we want to
+		// add pukless member, it has to be invite link. So old crypto
+		// members have to stay for now.
```

Comment:
We're stuck with old cryptomembers? That's bad security-wise. Is there a ticket to fix it?

Reply:
Old cryptomembers of a person will be sweeped by admin's SBS handler once person gets a PUK. I've just tried it, example change_membership created by SBS in impteam:
```
         "completed_invites": {
            "7fdae94b9a9b1abb1430da3c39187127": "7cc855e1457121c0c60188365b167d19%4"
         },
         "id": "a732f3e999ea054839a02739bb98f424",
         "is_implicit": true,
         "members": {
            "none": [
               "7cc855e1457121c0c60188365b167d19%1"
            ],
            "owner": [
               "7cc855e1457121c0c60188365b167d19%4"
            ]
         },
```